### PR TITLE
Fix incorrect vcd_vapp_vm 'network.x.is_primary' read when vCD NIC starts with index != 0

### LINKED ISSUE #458: vcd_vapp_vm: is_primary is not read correctly
Hello,
we have an VM such that:
- there is no nic 0
- it has a nic 1
- it has a nic 2

So we have two network interfaces, but counting starts with 1.
The second interface is primary

image

In this case, is_primary value is calculated wrong.

netIndex for primary is 1, but primaryNetworkConnectionIndex is 2:

https://github.com/terraform-providers/terraform-provider-vcd/blob/a7172f483940ed50ffeed9d4809c29d965e3ea1d/vcd/resource_vcd_vapp_vm.go#L1638

Instead, you need to compare networkConnectionIndex of the interface with primaryNetworkConnectionIndex

This changes worked for us on VCD 10

```
-       for netIndex, vmNet := range vm.VM.NetworkConnectionSection.NetworkConnection {
+       for _, vmNet := range vm.VM.NetworkConnectionSection.NetworkConnection { 
 [...]
-               if netIndex == vm.VM.NetworkConnectionSection.PrimaryNetworkConnectionIndex {
+               if vmNet.NetworkConnectionIndex == vm.VM.NetworkConnectionSection.PrimaryNetworkConnectionIndex { 
```

### Terraform Version
Terraform v0.12.18

### Affected Resource(s)
Please list the resources as a list, for example:
- vcd_vapp_vm

### Terraform Configuration Files
```hcl
  network {
    type               = "org"
    name               = var.network
    ip_allocation_mode = "POOL"
  }
  network {
    type               = "org"
    name               = var.network
    ip_allocation_mode = var.private_ip == null ? "POOL" : "MANUAL"
    ip                 = var.private_ip
    is_primary         = true
  }

```

### PR DESCRIPTION
**Pull request by Didainius**
This PR closes #458.
When vCD VMs have NIC definitions with indexes starting not from 0, `is_primary` property is assigned to incorrect network block and causes `updates` on refresh.

This PR introduces a test for it in commit 3828de7  and a fix in commit da8ec54.

Tested:
* `go test -v -tags ALL -run "TestAccVcdVappVm_NicIndex"` vCD 10.1 and 9.5
* `VCD_TEST_ORG_USER=1 go test -v -tags ALL -run "TestAccVcdVappVm_NicIndex"` vCD 10.1 and 9.5
* `make testvm`  vCD 10.1

### CODE REVIEW

File: vcd/resource_vcd_vapp_vm_test.go [hunk 1]
```diff
@@ -152,8 +160,163 @@ func TestAccVcdVAppVm_Clone(t *testing.T) {
...
+			if err != nil {
+				t.Errorf("error waiting for vApp deletion: %s", err)
+			}
+		}
+	}()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		// The step below imports an existing VM (created via go-vcloud-director SDK) with odd NIC
```
Comment (dataclouder) [inline vcd/resource_vcd_vapp_vm_test.go:201]: Any chance we can visualize the imported resource when GOVCD_DEBUG is set?

File: vcd/resource_vcd_vapp_vm_test.go [hunk 2]
```diff
@@ -152,8 +160,163 @@ func TestAccVcdVAppVm_Clone(t *testing.T) {
 	})
 }
 
-func init() {
-	testingTags["vm"] = "resource_vcd_vapp_vm_test.go"
+// TestAccVcdVappVm_NicIndex aims to replicate issue when VM is created outside of Terraform and
```
[the remark this answers is not in the document]
Comment (Didainius) [reply vcd/resource_vcd_vapp_vm_test.go]: Fixed.

File: vcd/resource_vcd_vapp_vm_test.go [hunk 3]
```diff
@@ -152,8 +160,163 @@ func TestAccVcdVAppVm_Clone(t *testing.T) {
...
+// NICs are not indexed starting with 0. It is not possible to create a VM with network cards
+// starting with index other than 0 in terraform-provider-vcd however it can happen when VM is
```
[the remark this answers is not in the document]
Comment (Didainius) [reply vcd/resource_vcd_vapp_vm_test.go]: Fixed.

File: vcd/resource_vcd_vapp_vm_test.go [hunk 4]
```diff
@@ -152,8 +160,163 @@ func TestAccVcdVAppVm_Clone(t *testing.T) {
...
+// imported from other systems.
+//
+// This test creates a VM using go-vcloud-director SDK with such vCD NIC indexes:
+// * vCD NIC index 1 (is_primary=false) - Terraform network block index 0 (is_primary=false)
+// * vCD NIC index 2 (is_primary=true) - Terraform network block index 1 (is_primary=true)
+// Before resolving below mentioned issue terraform-provider-vcd would report is_primary=false for
+// both NICs.
+// GitHub issue: https://github.com/terraform-providers/terraform-provider-vcd/issues/458
```
[the remark this answers is not in the document]
Comment (Didainius) [reply vcd/resource_vcd_vapp_vm_test.go]: Fixed

File: vcd/resource_vcd_vapp_vm_test.go (hunk 1 above)
Comment (Didainius) [reply vcd/resource_vcd_vapp_vm_test.go:203]: Added something like this when GOVCD_DEBUG is enabled:
```
=== RUN   TestAccVcdVappVm_NicIndex
computer_name = VM-001
cpu_cores = 1
cpus = 2
customization.# = 1
customization.0.allow_local_admin_password = true
customization.0.auto_generate_password = true
customization.0.change_sid = false
customization.0.enabled = false
customization.0.force = false
customization.0.join_domain = false
customization.0.join_org_domain = false
customization.0.must_change_password_on_first_login = false
customization.0.number_of_auto_logons = 0
description = created by test
disk.# = 0
expose_hardware_virtualization = false
guest_properties.% = 0
hardware_version = vmx-13
href = https://192.168.1.109/api/vApp/vm-a3e8e99a-dde5-4761-a51b-e9feee092fad
internal_disk.# = 1
internal_disk.0.bus_number = 0
internal_disk.0.bus_type = paravirtual
internal_disk.0.disk_id = 2000
internal_disk.0.iops = 0
internal_disk.0.size_in_mb = 16384
internal_disk.0.storage_profile = *
internal_disk.0.thin_provisioned = true
internal_disk.0.unit_number = 0
memory = 512
metadata.% = 0
name = TestAccVcdVappVm_NicIndex
network.# = 2
network.0.adapter_type = VMXNET3
network.0.ip_allocation_mode = NONE
network.0.is_primary = false
network.0.mac = 00:50:56:01:43:a9
network.0.type = none
network.1.adapter_type = VMXNET3
network.1.ip_allocation_mode = NONE
network.1.is_primary = true
network.1.mac = 00:50:56:01:43:aa
network.1.type = none
org = my-org
os_type = debian10Guest
override_template_disk.# = 0
storage_profile = *
vapp_name = TestAccVcdVappVm_NicIndex-vApp
vdc = my-org-vdc2

--- PASS: TestAccVcdVappVm_NicIndex (27.52s)
```

File: vcd/resource_vcd_vapp_vm_test.go [hunk 5]
```diff
@@ -152,8 +160,163 @@ func TestAccVcdVAppVm_Clone(t *testing.T) {
...
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		// The step below imports an existing VM (created via go-vcloud-director SDK) with odd NIC
+		// indexing and uses ImportStateCheck function to check that NIC priority is correctly set
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				ResourceName: "vcd_vapp_vm." + vmName + "-import",
+				ImportState:  true,
+				// ImportStateVerify: true,
```
[the remark this answers is not in the document]
Comment (Didainius) [reply vcd/resource_vcd_vapp_vm_test.go]: No, removed.

File: vcd/resource_vcd_vapp_vm_test.go [hunk 6]
```diff
@@ -152,8 +160,163 @@ func TestAccVcdVAppVm_Clone(t *testing.T) {
...
+				ImportStateIdFunc: importStateIdVappObject(testConfig, t.Name()+"-vApp", t.Name()),
+				ImportStateCheck:  validateNicPriority,
+			},
+		},
+	})
+}
+
+// validateNicPriority checks if NIC with vCD index is 2 is reported as primary nic after import
```
[the remark this answers is not in the document]
Comment (Didainius) [reply vcd/resource_vcd_vapp_vm_test.go]: Changed

File: vcd/resource_vcd_vapp_vm_test.go [hunk 7]
```diff
@@ -152,8 +160,163 @@ func TestAccVcdVAppVm_Clone(t *testing.T) {
...
+func validateNicPriority(state []*terraform.InstanceState) error {
+	// This imported VM has no NIC with index 0, but has NICs with indexes 1 and 2. However
+	// terraform starts indexing them from 0 therefore vCD NIC indexes do not coincide with
```
[the remark this answers is not in the document]
Comment (Didainius) [reply vcd/resource_vcd_vapp_vm_test.go]: Fixed.

**Pull request merged**